### PR TITLE
Add Trino 438 release notes

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 438 release.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 25 Jan 2024

* #20442 ✅ rn ✅ docs
* #20460 ✅ rn ✅ docs
* #20288 ✅ rn ✅ docs
* #20456 ✅ rn ✅ docs
* #20463 ✅ rn ✅ docs
* #20129 ✅ rn ✅ docs
* #20465 ✅ rn ✅ docs
* #20461 ✅ rn ✅ docs
* #20459 ✅ rn ✅ docs
* #20221 ✅ rn ✅ docs
* #20172 ✅ rn ✅ docs
* #20468 ✅ rn ✅ docs
* #20452 ✅ rn ✅ docs
* #20472 ✅ rn ✅ docs
* #20376 ✅ rn ✅ docs

## 26 Jan 2024

* #20464 ✅ rn ✅ docs
* #20470 ✅ rn ✅ docs
* #20475 ✅ rn ✅ docs
* #20462 ✅ rn ✅ docs
* #20366 ✅ rn ✅ docs

## 27 Jan 2024

* #20451 ✅ rn ✅ docs
* #20479 ✅ rn ✅ docs

## 28 Jan 2024

* #20488 ✅ rn ✅ docs

## 29 Jan 2024

* #20494 ✅ rn ✅ docs
* #20171 ✅ rn ✅ docs
* #20101 ✅ rn ✅ docs
* #20383 ✅ rn ✅ docs
* #20499 ✅ rn ✅ docs
* #17582 ✅ rn ✅ docs
* #20502 ✅ rn ✅ docs
* #20493 ✅ rn ✅ docs

## 30 Jan 2024

* #20505 ✅ rn ✅ docs
* #20496 ✅ rn ✅ docs
* #20380 ✅ rn ✅ docs
* #20512 ✅ rn ✅ docs
* #20514 ✅ rn ✅ docs
* #20506  ✅ rn ✅ docs
* #20173  ✅ rn ✅ docs
* #17779  ✅ rn ✅ docs
* #20388  ✅ rn ✅ docs
* #20519  ✅ rn ✅ docs

## 31 Jan 2024

* #20360 ✅ rn ✅ docs
* #20513 ✅ rn ✅ docs
* #20490 ✅ rn ✅ docs
* #20509 ✅ rn ✅ docs
* #20226 ✅ rn ✅ docs
* #20523 ✅ rn ✅ docs
* #19532 ✅ rn ✅ docs
* #20246 ✅ rn ✅ docs
* #20483 ✅ rn ✅ docs
* #20527 ✅ rn ✅ docs

## 1 Feb 2024

* #20522 ✅ rn ✅ docs
* #20498 ✅ rn ✅ docs
* #20535 ✅ rn ✅ docs
* #20536 ✅ rn ✅ docs
* #20540 ✅ rn ✅ docs
* #20538 ✅ rn ✅ docs
* #20529 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-438.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Release 438 (31 Jan 2024)
+
+## General
+
+* Add support for using `char` and `varchar` types with the {func}`reduce_agg`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-438.md:5]: I think this also applies to other types such as `uuid`, `ip_address`, `geometry`, `HyperLogLog`, etc.

Comment (wendigo) [reply docs/src/main/sphinx/release/release-438.md]: Yes. If only backed by the Slice

File: docs/src/main/sphinx/release/release-438.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Release 438 (31 Jan 2024)
+
+## General
+
+* Add support for using string types with the {func}`reduce_agg` function. ({issue}`20452`)
+* Fix query failure when using `char` types with the {func}`reverse` function. ({issue}`20387`)
+
+## Delta Lake connector
+
+* Add support for configuring the maximum value count for pages written in
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-438.md:10]: what is a "value count"

maybe 

the maximum pages

or 

the maximum number of pages

also the property name is similarly bad imho @martint 

and same change for all others

File: docs/src/main/sphinx/release/release-438.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## Hive connector
+
+* Add support for configuring the maximum value count for pages written in
+  Parquet files with the `parquet.writer.page-value-count` configuration
+  property or the `parquet_writer_page_value_count` session property. ({issue}`20171`)
+
+## Iceberg connector
+
+* Add support for the `ALTER COLUMN ... DROP NOT NULL` statement. ({issue}`20315`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-438.md:22]: Or plural .. whatever we use normally

```suggestion
* Add support for `ALTER COLUMN ... DROP NOT NULL` statements. ({issue}`20315`)
```

File: docs/src/main/sphinx/release/release-438.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+* Add support for configuring the maximum value count for pages written in
+  Parquet files with the `parquet.writer.page-value-count` configuration
+  property or the `parquet_writer_page_value_count` session property. ({issue}`20171`)
+* Add support for `array`, `map` and `row` types in the `migrate` table
+  procedure. ({issue}`17583`)
+* Fix potential query failure when running the `optimize` table procedure. ({issue}`)
+
+## Pinot connector
+
+* Add support for the `DATE` type. ({issue}`13059`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-438.md:32]: lowercase date?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-438.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-438.md:10]: I think it's the number of values that can be on a page, not the number of pages.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-438.md:10]: maybe.. its not clear to me ..

**Comment by mosabua** [comment]
Need to add #19532

File: docs/src/main/sphinx/release/release-438.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-438.md:10]: I will let @martint weigh in and decide.

**Review by mosabua — APPROVED** [review]
Looks good once the OPA stuff is in.

File: docs/src/main/sphinx/release/release-438.md (hunk 2 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-438.md:18]: This is a very low-level config. It appears to control the number of values that go in a parquet page within a column chunk. 

The current phrasing is the best we can do. Maybe change the grammar slightly to say: "the maximum number of values per page when writing to Parquet files"

File: docs/src/main/sphinx/release/release-438.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Release 438 (1 Feb 2024)
+
+## General
+
+* Add support for types such as `char`, `varchar`, `uuid`, `ip_address`,
+  `geometry`, and others with the {func}`reduce_agg` function. ({issue}`20452`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-438.md:6]: "[...] *to the* reduce_agg function" ?

File: docs/src/main/sphinx/release/release-438.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+* Fix query failure when using `char` types with the {func}`reverse` function. ({issue}`20387`)
+* Fix potential query failure when using the {func}`max_by` function on large
+  datasets. ({issue}`20524`)
+* Fix query failure when querying data with deeply-nested rows. ({issue}`20529`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-438.md:10]: The hyphen in "deeply-nested" isn't necessary and looks weird.

File: docs/src/main/sphinx/release/release-438.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+## Security
+
+* Add support for access control with
+  [Open Policy Agent](/security/opa-access-control). ({issue}`19532`)
+
+## Delta Lake connector
+
+* Add support for configuring the maximum number of values per page when writing
+  to Parquet files with the `parquet.writer.page-value-count` configuration
+  property or the `parquet_writer_page_value_count` session property. ({issue}`20171`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-438.md:21]: The entry for `ALTER COLUMN ... DROP NOT NULL` is missing. https://github.com/trinodb/trino/pull/20491

**Pull request merged**